lib/Bio/EnsEMBL/DataCheck/Checks/CheckTableSizes.pm

```diff
@@ -40,7 +40,10 @@ sub tests {
   
   my $curr_dba = $self->dba;
   my $curr_helper = $curr_dba->dbc->sql_helper;
-  my $prev_dba = $self->registry->get_DBAdaptor('compara_prev', 'compara') || $self->get_old_dba;
+  my $prev_dba;
+  $prev_dba = $self->registry->get_DBAdaptor('compara_prev', 'compara') if $self->registry_file;
+  $prev_dba = $self->get_old_dba if $self->old_server_uri && !$prev_dba;
+  die "Neither -registry_file nor -old_server_uri given on the command-line. Cannot find the previous database" unless $prev_dba;
```

Comment:
Generally speaking, it's better to throw rather than die in a hive module. (But are you sure you want to die here? If this datacheck is being run as part of a group of datachecks, that will abort all of them. I know I've previously said it's best to avoid using `fail` directly, but rules are there to be broken...)
And since this could be run within a pipeline, might be better to remove reference to the command line, to avoid confusion.
```suggestion
  $self->throw("Neither 'registry_file' nor 'old_server_uri' parameter given. Cannot find the previous database") unless $prev_dba;
```

Reply:
Right, good point. I think I would prefer using `fail` and doing an early `return` instead.

Reply:
By the way, there are dozens of `die` statements in the repo (BaseCheck.pm, DbCheck.pm, Checks/*.pm)

Reply:
Ah, yes... It's OK in BaseCheck and DbCheck, because those are part of the infrastructure; if something is wrong there, then you probably do want everything to fall down around your ears. I'll create a ticket to look into the Checks/*.pm though. Thanks!